cognite_toolkit/_cdf_tk/utils/table_writers.py

```diff
@@ -0,0 +1,243 @@
+            self._writer_by_filepath[file_path] = self._create_writer(file_path)
+        elif self._is_above_file_size_limit(file_path, self._writer_by_filepath[file_path]):
+            self._writer_by_filepath[file_path].close()
+            del self._writer_by_filepath[file_path]
+            self._file_count += 1
+            return self._get_writer(group)
+        return self._writer_by_filepath[file_path]
+
+    @classmethod
+    def get_write_cls(cls, format_: FileFormat) -> "type[TableFileWriter]":
+        """Get the writer class for the given format."""
+        write_cls = _TABLEWRITER_CLASS_BY_FORMAT.get(format_)
+        if write_cls is None:
+            raise ToolkitValueError(
+                f"Unsupported format {format_}. Supported formats are {humanize_collection(_TABLEWRITER_CLASS_BY_FORMAT.keys())}."
+            )
+        return write_cls
+
+
+class ParquetWriter(TableFileWriter["pq.ParquetWriter"]):
+    format = "parquet"
+
+    def __init__(self, schema: Schema, output_dir: Path, max_file_size_bytes: int = 128 * 1024 * 1024) -> None:
+        super().__init__(schema, output_dir, max_file_size_bytes)
+        self._check_pyarrow_dependency()
+
+    def _create_writer(self, filepath: Path) -> "pq.ParquetWriter":
+        import pyarrow.parquet as pq
+
+        schema = self._create_schema()
+        return pq.ParquetWriter(filepath, schema)
+
+    def _write_rows(self, writer: "pq.ParquetWriter", rows: Rows) -> None:
+        import pyarrow as pa
+
+        if json_columns := self._json_columns():
+            for row in rows:
+                json_values = set(row.keys()) & json_columns
+                for col in json_values:
+                    row[col] = json.dumps(row[col])
+
+        table = pa.Table.from_pylist(rows, schema=self._create_schema())
+        writer.write_table(table)
+
+    def _is_above_file_size_limit(self, filepath: Path, writer: "pq.ParquetWriter") -> bool:
+        return filepath.exists() and filepath.stat().st_size > self.max_file_size_bytes
+
+    @lru_cache(maxsize=1)
+    def _json_columns(self) -> set[str]:
+        """Check if the writer supports JSON format."""
+        return {col.name for col in self.schema.columns if col.type == "json"}
+
+    @lru_cache(maxsize=1)
+    def _create_schema(self) -> "pa.Schema":
+        """Create a pyarrow schema from the schema definition."""
+        self._check_pyarrow_dependency()
+        import pyarrow as pa
+
+        fields: list[pa.Field] = []
+        for prop in self.schema.columns:
+            pa_type = self._as_pa_type(prop.type, prop.is_array)
+            fields.append(pa.field(prop.name, pa_type, nullable=True))
+        return pa.schema(fields)
+
+    @staticmethod
+    def _check_pyarrow_dependency() -> None:
+        if importlib.util.find_spec("pyarrow") is None:
+            raise ToolkitMissingDependencyError(
+                "Writing to parquet requires pyarrow. Install with 'pip install cognite-toolkit[table]'"
```

Comment:
This fails in some terminals, enclosing in single or double quotes may be necessary: 

```suggestion
                "Writing to parquet requires pyarrow. Install with 'pip install \"cognite-toolkit[table]\"'"
```